### PR TITLE
Fix dashboard test bug

### LINKED ISSUE #19503: [Bug] Dashboard tests are not properly tested
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Dashboard

### What happened + What you expected to happen

Currently, wait_for_condition that's used for the dashboard tests only return True(succeed)/False(not succeed). That says if you don't put the assert, it won't actually test things properly unlike `wait_for_condition` (which raises an exception if it doesn't succeed).

It seems like this was the case for many dashboard tests. When I tried enabling some of them, they failed (meaning there were regressions for a long time).

NOTE:
https://github.com/ray-project/ray/pull/19385#discussion_r729095611

### Versions / Dependencies

Master

### Reproduction script

Check https://github.com/ray-project/ray/pull/19385#discussion_r729095611

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently `wait_until_succeeded_without_exception` is used in the dashboard, and it returns True/False. Unfortunately, there are lots of code that doesn't assert on this method (which means things are not actually tested). 

## Related issue number

Closes https://github.com/ray-project/ray/issues/19503

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

File: dashboard/modules/node/tests/test_node.py [hunk 1]
```diff
@@ -489,58 +489,5 @@ def check_logs():
         check_logs, (AssertionError, ), timeout_ms=10000)
 
 
-@pytest.mark.parametrize(
-    "ray_start_cluster_head", [{
-        "include_dashboard": True
-    }], indirect=True)
-def test_errors(enable_test_module, disable_aiohttp_cache,
```
Comment (rkooo567) [inline dashboard/modules/node/tests/test_node.py:496]: This feature is not working anymore since we don't log exceptions to log files anymore (instead we handle this from the caller).

But I found this problem; https://github.com/ray-project/ray/issues/21810

File: dashboard/memory_utils.py [hunk 2]
```diff
@@ -390,8 +394,8 @@ def memory_summary(state,
         "IP Address", "PID", "Type", "Call Site", "Size", "Reference Type",
         "Object Ref"
     ]
-    object_ref_string = "{:<8} | {:<3} | {:<4} | {:<9} \
-| {:<4} | {:<14} | {:<10}\n"
+    object_ref_string = "{:<13} | {:<8} | {:<7} | {:<9} \
```
Comment (rkooo567) [inline dashboard/memory_utils.py:397]: Made it longer because it will have better display. Idk why it was so short

File: dashboard/memory_utils.py [hunk 3]
```diff
@@ -340,7 +340,11 @@ def track_reference_size(group):
         "ACTOR_HANDLE": "total_actor_handles"
     }
     for entry in group["entries"]:
-        d[table_name[entry["reference_type"]]] += entry["object_size"]
+        size = entry["object_size"]
```
Comment (rkooo567) [inline dashboard/memory_utils.py:343]: This was another bug that is unrelated to this PR. This makes the size of actor handle negative.

**Review by edoakes — APPROVED** [review]
Thanks!
